bigbluebutton-html5/imports/ui/components/whiteboard/whiteboard-toolbar/component.jsx

```diff
@@ -1 +1,534 @@
+    this.handleSwitchWhiteboardMode = this.handleSwitchWhiteboardMode.bind(this);
+    this.handleAnnotationChange = this.handleAnnotationChange.bind(this);
+    this.handleThicknessChange = this.handleThicknessChange.bind(this);
+    this.handleFontSizeChange = this.handleFontSizeChange.bind(this);
+    this.handleColorChange = this.handleColorChange.bind(this);
+    this.disableOnBlur = this.disableOnBlur.bind(this);
+    this.enableOnBlur = this.enableOnBlur.bind(this);
+    this.renderAnnotationList = this.renderAnnotationList.bind(this);
+    this.renderFontSizeList = this.renderFontSizeList.bind(this);
+    this.renderThicknessList = this.renderThicknessList.bind(this);
+    this.renderColorList = this.renderColorList.bind(this);
+  }
+
+  componentWillMount() {
+    // setting default or resetting current drawing settings in the session
+    const { annotationSelected, thicknessSelected, colorSelected, fontSizeSelected } = this.state;
+    this.props.actions.setWhiteboardToolbarValues(
+      annotationSelected.sessionValue,
+      thicknessSelected.sessionRadius,
+      WhiteboardToolbar.HEXToINTColor(colorSelected),
+      fontSizeSelected,
+      {
+        textShapeValue: '',
+        textShapeActiveId: '',
+      },
+    );
+  }
+
+  componentDidMount() {
+    // to let the whiteboard know that the presentation area's size has changed
+    window.dispatchEvent(new Event('resize'));
+
+    if (this.state.annotationSelected.sessionValue !== 'text') {
+      // trigger initial animation on the thickness circle, otherwise it stays at 0
+      this.thicknessListIconRadius.beginElement();
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    // if color or thickness were changed
+    // we might need to trigger svg animation for Color and Thickness icons
+    this.animateSvgIcons(prevState);
+  }
+
+  componentWillUnmount() {
+    // to let the whiteboard know that the presentation area's size has changed
+    window.dispatchEvent(new Event('resize'));
+  }
+
+  animateSvgIcons(prevState) {
+    /* Animation for the svg icons that we use for thickness (circle) and color (rectangle)
+     * has to be triggered manually
+     * we have 4 main cases:
+     * 1. Color change -
+         a) Text tool is selected, Font-Size icon substitutes the thickness icon,
+            thus we need to trigger just color change for the color icon
+         b) Any other tool than Text tool is selected - trigger color change for both icons
+     * 2. Thickness change - trigger radius for the thickness icon
+     * 3. Switch from the Text tool to any other - trigger color and radius for thickness
+     * 4. Trigger initial animation for the icons
+    */
+
+    // 1st case
+    if (this.state.colorSelected !== prevState.colorSelected) {
+      // 1st case a)
+      if (this.state.annotationSelected.sessionValue === 'text') {
+        this.colorListIconColor.beginElement();
+      // 1st case b)
+      } else {
```

Comment:
Else is unnecessary, this might became

```
if (this.state.annotationSelected.sessionValue !== 'text') {
 this.thicknessListIconColor.beginElement();
}
this.colorListIconColor.beginElement();
```